unittests/test_hochfrequenz_condition_resolver.py

```diff
@@ -0,0 +1,99 @@
+
+from ahbichtfunctionsclient import HochfrequenzConditionResolver
+
+pytestmark = pytest.mark.asyncio
+
+
+class TestHochfrequenzConditionResolver:
+    async def test_hochfrequenz_condition_api_success(self):
+        condition_resolver = HochfrequenzConditionResolver(
+            edifact_format_version=EdifactFormatVersion.FV2204,
+            edifact_format=EdifactFormat.UTILMD,
+            api_url="https://test.inv",
+        )
+        with aioresponses() as mocked_server:
+            mocked_server.get(
+                "https://test.inv/FV2204/UTILMD/10P",
+                payload={
+                    "condition_text": "Wenn foo[56] und bar[99] dann FooBar.",
+                    "condition_key": "10P",
+                    "edifact_format": "UTILMD",
+                },
+            )
+            actual = await condition_resolver.get_condition_text("10P")
+            assert actual == ConditionKeyConditionTextMapping(
+                edifact_format=EdifactFormat.UTILMD,
+                condition_key="10P",
+                condition_text="Wenn foo[56] und bar[99] dann FooBar.",
+            )
+
+    async def test_hochfrequenz_condition_api_failure(self):
+        condition_resolver = HochfrequenzConditionResolver(
+            edifact_format_version=EdifactFormatVersion.FV2204,
+            edifact_format=EdifactFormat.UTILMD,
+            api_url="https://test.inv",
+        )
+
+        def simulate_error(url, **kwargs):
+            return CallbackResult(status=400, payload={"it is not": "important what's here, just that you had to wait"})
+
+        with aioresponses() as mocked_server:
+            mocked_server.get(url="https://test.inv/FV2204/UTILMD/001", callback=simulate_error, repeat=5)
+            actual = await condition_resolver.get_condition_text("001")
+            assert actual == ConditionKeyConditionTextMapping(
+                # see the documentation: if the package could not be resolved, you'll get a None package_expression
+                # but the PackageKeyConditionExpressionMapping itself is _not_ None
+                edifact_format=EdifactFormat.UTILMD,
+                condition_key="001",
+                condition_text=None,
+            )
+
+    async def test_async_behaviour(self):
+        condition_resolver = HochfrequenzConditionResolver(
+            edifact_format_version=EdifactFormatVersion.FV2204,
+            edifact_format=EdifactFormat.UTILMD,
+            api_url="https://test.inv",
+        )
+
+        async def wait_some_time(url, **kwargs):
+            await asyncio.sleep(1)
+            return CallbackResult(status=400, payload={"it is not": "important what's here, just that you had to wait"})
+
+        with aioresponses() as mocked_server:
+            tasks = []
+            for x in range(1, 6):
+                mocked_server.get(url=f"https://test.inv/FV2204/UTILMD/{x}P", callback=wait_some_time)
+                tasks.append(condition_resolver.get_condition_text(f"{x}P"))
```

Comment:
```suggestion
            for x in range(1, 6):
                mocked_server.get(url=f"https://test.inv/FV2204/UTILMD/{x}", callback=wait_some_time)
                tasks.append(condition_resolver.get_condition_text(f"{x}"))
```
anmalog oben: realistische testdaten verwenden.